services/dy-3dvis/tests/integration/test_docker_container.py

```diff
@@ -0,0 +1,148 @@
+    env = {"{}_FOLDER".format(str(folder).upper()): (
+        _CONTAINER_FOLDER / folder).as_posix() for folder in _FOLDER_NAMES}
+    return env
+
+
+@pytest.fixture
+def validation_folders(validation_dir: Path) -> Dict:
+    return {folder: (validation_dir / folder) for folder in _FOLDER_NAMES}
+
+
+@pytest.fixture
+def docker_container(validation_folders: Dict, host_folders: Dict, docker_client: docker.DockerClient, docker_image_key: str, container_variables: Dict) -> docker.models.containers.Container:
+    # copy files to input folder, copytree needs to not have the input folder around.
+    host_folders["input"].rmdir()
+    shutil.copytree(validation_folders["input"], host_folders["input"])
+    assert Path(host_folders["input"]).exists()
+    # run the container (this may take some time)
+    try:
+        volumes = {host_folders[folder]: {"bind": container_variables["{}_FOLDER".format(
+            str(folder).upper())]} for folder in _FOLDER_NAMES}
+        container = docker_client.containers.run(docker_image_key,
+                                                 "run", detach=True, remove=False, volumes=volumes, environment=container_variables)
+        response = container.wait()
+        if response["StatusCode"] > 0:
+            logs = container.logs(timestamps=True)
+            pytest.fail("The container stopped with exit code {}\n\n\ncommand:\n {}, \n\n\nlog:\n{}".format(response["StatusCode"],
+                                                                                                            "run", pformat(
+                (container.logs(timestamps=True).decode("UTF-8")).split("\n"), width=200
+            )))
+        else:
+            yield container
+    except docker.errors.ContainerError as exc:
+        # the container did not run correctly
+        pytest.fail("The container stopped with exit code {}\n\n\ncommand:\n {}, \n\n\nlog:\n{}".format(exc.exit_status,
+                                                                                                        exc.command, pformat(
+                                                                                                            (container.logs(timestamps=True).decode("UTF-8")).split("\n"), width=200
+                                                                                                        )))
+    finally:
+        # cleanup
+        container.remove()
+
+
+def _convert_to_simcore_labels(image_labels: Dict) -> Dict:
+    io_simcore_labels = {}
+    for key, value in image_labels.items():
+        if str(key).startswith("io.simcore."):
+            simcore_label = json.loads(value)
+            simcore_keys = list(simcore_label.keys())
+            assert len(simcore_keys) == 1
+            simcore_key = simcore_keys[0]
+            simcore_value = simcore_label[simcore_key]
+            io_simcore_labels[simcore_key] = simcore_value
+    assert len(io_simcore_labels) > 0
+    return io_simcore_labels
+
+
```

Comment:
wanna keep for something? if so, please add note otherwise delete

Reply:
yes you are right... I kept it for later but it's true. until the dy-sidecar is around...